Front/addcategory.cpp

```diff
@@ -0,0 +1,31 @@
+#include "addcategory.h"
+#include "ui_addcategory.h"
+#include "category.h"
+
+AddCategory::AddCategory(QWidget *parent) :
+    QDialog(parent),
+    ui(new Ui::AddCategory)
+{
+    ui->setupUi(this);
+    categoryform = new Category();
```

Comment:
Memory leak, add `delete` in the destructor or better use smart pointer

Reply:
Fixed